test/specs/upload/tests/02.upload_obj.js

```diff
@@ -0,0 +1,262 @@
+                    if (chunks[i].completed) {
+                        expect(chunks[i].progress).toBe(chunkSize, "Chunk progress is not the same as chunk size after completion");
+                        numCompleteChunks++;
+                    } else {
+                        expect(chunks[i].progress).toBe(0, "Chunk progress has started even though it's not complete");
+                        numIncompleteChunks++;
+                    }
+                }
+                expect(numCompleteChunks).toBe(counter, "Not enough chunks completed");
+                expect(numIncompleteChunks).toBe(chunks.length-counter, "Too many chunks completed");
+
+                // verify the file is uploaded or not
+                if(counter == chunks.length) {
+                    expect(uploadObj.completed).toBeTruthy("File not yet created");
+                } else {
+                    expect(uploadObj.completed).toBeFalsy("File has been created");
+                }
+                counter++;
+            }).catch(function(err) {
+                console.dir(err);
+                done.fail();
+            });
+        });
+
+        it("should complete the upload.", function (done) {
+            uploadObj.completeUpload().then(function(response) {
+                expect(response.startsWith(serverFilePath)).toBeTruthy("Upload job file path is incorrect");
+                expect(uploadObj.jobDone).toBeTruthy("Upload job is not complete");
+
+                done();
+            }, function(err) {
+                console.dir(err);
+                done.fail();
+            });
+        });
+
+        it("should create another upload job, set it to paused, and resume the upload.", function (done) {
+            var anotherUploadObj = new options.ermRest.Upload(file.file, {
+                column: column,
+                reference: reference,
+                chunkSize: chunkSize
+            });
+
+            anotherUploadObj.calculateChecksum(validRow).then(function() {
+
+                return anotherUploadObj.createUploadJob();
+            }).then(function(response) {
+                expect(response.startsWith(serverFilePath)).toBeTruthy("Upload job file path is incorrect");
+                anotherUploadObj.isPaused = true;
+
+                return anotherUploadObj.resume();
+            }).then(function(response) {
+                expect(anotherUploadObj.isPaused).toBeFalsy("Upload job is not paused");
+
+                return uploadObj.completeUpload();
+            }).then(function(response) {
+                expect(response).toBe(serverFilePath, "Server file path is incorrect");
+                expect(uploadObj.jobDone).toBeTruthy("Upload job is not complete");
+
+                done();
+            }).catch(function(err) {
+                console.dir(err);
+                done.fail();
+            });
+        });
+
+        // TODO: Upload.pause(), Upload.resume(), and Upload.cancel() are not tested.
```

Comment:
Can you add some comments that why we are not testing these function now, and how we might be able to test them later?